File: src/ui/src/main/java/com/amazon/sample/ui/services/checkout/model/ShippingAddress.java
package com.amazon.sample.ui.services.checkout.model;

import lombok.Data;

@Data
public class ShippingAddress {
    private String firstName;

    private String lastName;

    private String email;

    private String address1;

    private String address2;

    private String city;

    private String zip;

    private String state;
}

File: src/checkout/src/checkout/models/ShippingAddress.ts
import { IsOptional, IsString } from 'class-validator';

export class ShippingAddress {

  @IsString()
  firstName : string;

  @IsString()
  lastName : string;

  @IsString()
  address1 : string;

  @IsString()
  @IsOptional()
  address2 : string;

  @IsString()
  city : string;

  @IsString()
  state : string;

  @IsString()
  zip : string;

}